src/remote_control.cpp

```diff
@@ -158,51 +147,34 @@ bool RemoteControl::waitForNextStepCommon(const std::string& caption, bool auton
   {
     std::stringstream ss;
     ss << CONSOLE_COLOR_CYAN << "Waiting to continue: " << caption << CONSOLE_COLOR_RESET;
-    RCLCPP_ERROR(logger_, ss.str().c_str());
+    RCLCPP_INFO(logger_, ss.str().c_str());
   }
 
   if (displayWaitingState_)
   {
     displayWaitingState_(true);
   }
 
-  next_step_ready_ = std::make_unique<std::promise<void>>();
+  is_waiting_ = true;
   // Wait until next step is ready
-  std::shared_future<void> future_next_step_ready = next_step_ready_->get_future();
-  while (!autonomous && rclcpp::ok())
+  while (!next_step_ready_ && !autonomous && rclcpp::ok())
   {
-    /* TODO(mlautman): Pending https://github.com/ros2/rclcpp/issues/520 the only way to spin is by
-     * having access to the executor. Thus, the remote control must have an executor. Once the issue
-     * has been resolved, it would be nice to remove the executor from this class */
-    rclcpp::FutureReturnCode status = executor_->spin_until_future_complete(
-        future_next_step_ready, std::chrono::milliseconds(125));
-    if (status == rclcpp::FutureReturnCode::SUCCESS)
-    {
-      break;
-    }
-    else if (status == rclcpp::FutureReturnCode::INTERRUPTED)
-    {
-      RCLCPP_INFO(logger_, "Spinning was interrupted.");
-      next_step_ready_ = nullptr;
-      if (!rclcpp::ok())
-      {
-        exit(0);
-      }
-    }
+    rclcpp::sleep_for(125ms);
```

Comment:
You can comfortably reduce this to 10ms, the CPU used will be ridiculously low anyway.

If you want to be very elegant, use [std::condition_variable](https://en.cppreference.com/w/cpp/thread/condition_variable) instead of sleep.